docs/start-date-computation.md

```diff
@@ -0,0 +1,74 @@
+
+And last, but not least, let's assume that today is 07th March 2024.
+
+### A note on anniversaries.
+
+Imagine that you work for HR and are given the task of giving a gift of the thousands of people working for a company. You can only give to somebody their gift on their birthday. It's going to take you an entire year to do so. If today is the 7th of March 2024, then you will give gifts to a bunch of people today (all the people with a 7th of March birthday) and you will finish on the 6th of March 2025.
+
+If you are then told to not use the yearly birthday, but simply the number of their birthdate, so for instance if somebody is born on Dec 23rd, you can give them their gift on 23rd of March. Then, it will take you one month to give all the gifts away. You will start today on 7th of March and will finish on 6th of April.
+
+But then you realise that if you use the monthly anniversary of the person, then there are many more people to give a gift to per day (exactly 12 times more people actually), and you can't run that fast. So you decide of a spread period. You decide that you will do it over three months. So you attribute a random number from the set { 0, 1, 2 } to each person on your distribution spreadsheet, and if the number is 0 you give them the gift the next available day number, otherwise you postpone by that random choice number of months.
+
+### Spread periods
+
+When Marketing says that they want to spead the monthlies over 3 months, they mean that the process of upgrading the monthlies to a new price (which should otherwise only need 1 month) should be planned to take 3 months. You know that there will be a random choice made as part of the decision of the subscription start date. 
+
+We could, but we never use a spread period for other billing frequencies (quarterly, annual, etc)
+
+### Why do we use spread periods
+
+If we have a lot of monthlies to migrate, that's a lot of user notifications per day and therefore a certain number of people calling the Guardian call centers per day. By spreading the monthlies over 3 months instead of 1 month, we help the call center operators with not being flooded with daily calls.
+
+### Computing the start date.
+
+With the above context and the explanation about spread periods, let's compute the start date of our subscription.
+
+Step 1: We know that the chosen date needs to be after the cohort spec's `earliestPriceMigrationStartDate`, meaning after `2024-05-20`. The date `2024-05-20` is our first lowerbound.
+
+Step 2: We know that the date needs to be after today plus the end of a notification period (otherwise the engine will alarm immediately). The notification period is `[-49, -36]`, so there should be at least 37 days between today and the chosen date. This means thay we have a lowerbound at `today + 37 days`, meaning `2024-03-07 + 37 days`, meaning `2024-04-13` (13th of April).
```

Comment:
```suggestion
Step 2: We know that the date needs to be after today plus the end of a notification period (otherwise the engine will alarm immediately). Remember we are using `2024-03-07` as "today". The notification period is `[-49, -36]`, so there should be at least 37 days between today and the chosen date. This means thay we have a lowerbound at `today + 37 days`, meaning `2024-03-07 + 37 days`, meaning `2024-04-13` (13th of April).
```